src/EditorFeatures/Core.Wpf/Completion/Presentation/CompletionPresenterSession.cs

```diff
@@ -132,6 +135,7 @@ internal sealed class CompletionPresenterSession : ForegroundThreadAffinitizedOb
                 // so that we can call back into ourselves, get the items and add it to the
                 // session.
                 _editorSessionOpt.Properties.AddProperty(Key, this);
+                _editorSessionOpt.Properties.AddProperty(DocumentKey, document);
```

Comment:
for the old, synchronous, completoin system, we just store the Document on the ephemeral ICompletionSession.  Note: we don't need to do this for the new, async, completion service.  In that world, the IAsyncCompletionSession has kept track of the trigger snapshot, and we use that to get back to the Document in question.